modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/IgniteCacheNearRestartRollbackSelfTest.java

```diff
@@ -166,13 +174,15 @@ public void testRestarts() throws Exception {
                 catch (Throwable e) {
                     log.error("Update failed: " + e, e);
 
+                    doRestarts.set(false);
+
                     throw e;
                 }
             }
-
-            fut.get();
         }
         finally {
+            fut.get();
```

Comment:
What if fut.get() will throw an exception for some reason?
It's anyway better to stop grids in afterTest() to avoid any risks in test method code.

Reply:
got it